### DISCUSSION TITLE
Statement executeReturning

### DISCUSSION THREAD

**Discussion by marcin-bury** [asfernandes/node-firebird-drivers#111] [Q&A]
Hi

I'd like to ask if this is a bug or "by desing" behavior.

When I'm trying tu execute "executeReturning" for sql statement that returns empty resultset, I'm always  getting "attempt to fetch past the last record in a record stream".
Should I add some execute options to get empty array?

Thanks,
Marcin

**Comment by asfernandes** [comment]
```
/** Executes a statement that returns a single record as [col1, col2, ..., colN]. */
```

It should not mix empty record with list of columns, so I do think it's correct.

**Reply by marcin-bury** [reply]
So expecting an empty result I should use result object, right?

**Comment by asfernandes** [comment]
executeQuery

**Comment by asfernandes** [comment]
I'm thinking in add some method that returns query as array of records directly instead of need to get them with result set.
